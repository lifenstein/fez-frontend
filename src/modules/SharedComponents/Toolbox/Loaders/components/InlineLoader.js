import React from 'react';
import PropTypes from 'prop-types';
import CircularProgress from '@material-ui/core/CircularProgress';
import Grid from '@material-ui/core/Grid';
import Hidden from '@material-ui/core/Hidden';
import Typography from '@material-ui/core/Typography';
import { withStyles } from '@material-ui/core/styles';

export class InlineLoader extends React.Component {
    static propTypes = {
        message: PropTypes.string,
        classes: PropTypes.object,
        loaderId: PropTypes.string,
    };

    static defaultProps = {
        message: 'Loading',
    };

    render() {
        const { loaderId, message } = this.props;
        return (
            <div style={{ padding: 8 }}>
                <Grid
                    container
                    direction={'row'}
                    spacing={1}
                    justify="center"
                    alignItems="center"
                    alignContent={'center'}
                    id={loaderId}
                    data-testid={loaderId}
                >
                    <Hidden smUp>
                        <Grid item xs />
                    </Hidden>
                    <Grid item xs={'auto'} style={{ textAlign: 'center' }}>
                        <CircularProgress size={18} thickness={2} color="primary" />
                    </Grid>
                    <Grid item xs={'auto'} style={{ textAlign: 'center' }}>
                        <Typography color={'primary'} variant={'h5'} component={'span'} style={{ fontSize: '1.33rem' }}>
                            {message}
                        </Typography>
                    </Grid>
                    <Hidden smUp>
                        <Grid item xs />
                    </Hidden>
                </Grid>
            </div>
        );
    }
}

export default withStyles(null, { withTheme: true })(InlineLoader);
