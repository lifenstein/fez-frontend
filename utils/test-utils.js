/* eslint-disable react/prop-types */
import React from 'react';
import PropTypes from 'prop-types';
import { render } from '@testing-library/react';
import { Router } from 'react-router-dom';
import { Route } from 'react-router';
import { mui1theme } from 'config/theme';
import { Provider } from 'react-redux';
import { ThemeProvider as MuiThemeProvider } from '@material-ui/core/styles';
import { MuiPickersUtilsProvider } from '@material-ui/pickers';
import MomentUtils from '@date-io/moment';

import { getStore } from '../src/config/store';
import Immutable from 'immutable';
import { createMemoryHistory } from 'history';

const domTestingLib = require('@testing-library/dom');
const reactTestingLib = require('@testing-library/react');
const mime = require('mime-types');

const { configure } = domTestingLib;

configure(config => ({
    ...config,
    testIdAttribute: 'id',
}));

export const AllTheProviders = props => {
    return (
        <MuiThemeProvider theme={mui1theme}>
            <MuiPickersUtilsProvider utils={MomentUtils}>{props.children}</MuiPickersUtilsProvider>
        </MuiThemeProvider>
    );
};

AllTheProviders.propTypes = {
    children: PropTypes.node,
};

export const rtlRender = (ui, options) => render(ui, { wrapper: AllTheProviders, ...options });

export const renderWithRouter = (
    ui,
    { route = '/', history = createMemoryHistory({ initialEntries: [route] }), renderMethod = render } = {},
) => {
    return renderMethod(
        <AllTheProviders>
            <Router history={history}>{ui}</Router>
        </AllTheProviders>,
    );
};

export const WithRouter = ({ children, route = '/', history = createMemoryHistory({ initialEntries: [route] }) }) => (
    <AllTheProviders>
        <Router history={history}>{children}</Router>
    </AllTheProviders>
);
export const renderWithRedux = ({ initialState }) => render => {
    return {
        ...render,
        store: getStore({ initialState, history: render.history }),
    };
};

export const withRouter = ({
    route = '/',
    path = '/',
    history = createMemoryHistory({ initialEntries: [route] }),
} = {}) => WrappedComponent => {
    return (
        <Router history={history}>
            <Route path={path} children={WrappedComponent} />
        </Router>
    );
};

export const withRedux = (initialState = Immutable.Map()) => WrappedComponent => {
    return <Provider store={getStore(initialState)}>{WrappedComponent}</Provider>;
};

export const WithReduxStore = ({ initialState = Immutable.Map(), children }) => (
    <Provider store={getStore(initialState)}>
        <AllTheProviders>{children}</AllTheProviders>
    </Provider>
);

export const defaultDatastream = {
    dsi_embargo_date: null,
    dsi_open_access: 1,
    dsi_label: '',
    dsi_copyright: null,
    dsi_state: 'A',
    dsi_size: 27932352,
};
export const createFezDatastreamInfoArray = (datastreams, pid = null, withPreview = true) => {
    let processed = datastreams.filter(datastream => typeof datastream === 'object' || typeof datastream === 'string');
    if (withPreview) {
        processed = [];
        datastreams.forEach(function(datastream) {
            processed.push(datastream);
            const filename = typeof datastream === 'object' ? datastream.dsi_dsid : datastream;
            const mimetype = mime.lookup(filename);
            // bail in case it's not derivable
            if (
                !filename.includes('.pdf') &&
                !mimetype.includes('image') &&
                !mimetype.includes('audio') &&
                !mimetype.includes('video')
            ) {
                return;
            }
            // eslint-disable-next-line no-nested-ternary
            const derivativeMimetype = mimetype.includes('audio')
                ? 'audio/mp3'
                : mimetype.includes('video')
                ? 'video/mp4'
                : 'image/jpg';

            (derivativeMimetype.includes('image') ? ['preview_', 'thumbnail_', 'web_'] : ['']).forEach(function(
                prefix,
            ) {
                processed.push({
                    dsi_dsid: `${prefix}${filename.split('.')[0]}_t.${derivativeMimetype.split('/').pop()}`,
                    dsi_mimetype: derivativeMimetype,
                });
            });
        });
    }

    return processed.map((datastream, index) => ({
        ...defaultDatastream,
        ...{
            dsi_pid: pid,
            dsi_order: index + 1,
            dsi_mimetype: mime.lookup(typeof datastream === 'object' ? datastream.dsi_dsid : datastream),
        },
        ...(typeof datastream === 'object'
            ? datastream
            : {
                  dsi_dsid: datastream,
              }),
    }));
};

export const withDatastreams = (sources, datastreams, callback) =>
    datastreams.map(item => {
        const source = sources.find(datastream =>
            datastream.dsi_dsid.includes(
                item.dsi_dsid
                    .replace('_t.', '.')
                    .replace('preview_', '')
                    .replace('thumbnail_', '')
                    .replace('web_', '')
                    .split('.')[0],
            ),
        );
        const isDerivative = item.dsi_dsid !== source.dsi_dsid;
        callback(item, source, isDerivative);
    });

export const getDatastreamByFilename = (filename, derivatives) =>
    derivatives.find(derivative => derivative.dsi_dsid === filename);

module.exports = {
    ...domTestingLib,
    ...reactTestingLib,
    rtlRender,
    renderWithRouter,
    renderWithRedux,
    withRedux,
    withRouter,
    AllTheProviders,
    WithReduxStore,
    WithRouter,
    createFezDatastreamInfoArray,
    getDatastreamByFilename,
    withDatastreams,
};
