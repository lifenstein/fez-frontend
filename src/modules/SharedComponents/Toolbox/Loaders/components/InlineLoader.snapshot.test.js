import React from 'react';
import { InlineLoader } from './InlineLoader';
import { rtlRender } from 'test-utils';

function setup(testProps = {}) {
    const props = {
        ...testProps,
        classes: {
            text: {
                fontWeight: 200,
                margin: '24px 0',
            },
        },
    };
    return rtlRender(<InlineLoader {...props} />);
}

describe('Component InlineLoader', () => {
    it('should render as expected', () => {
        const props = {
            message: 'This is a tst',
        };
        const { container } = setup({ ...props });
        expect(container).toMatchSnapshot();
    });
});
