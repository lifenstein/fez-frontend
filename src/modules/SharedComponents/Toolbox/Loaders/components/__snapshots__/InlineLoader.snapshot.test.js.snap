// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component InlineLoader should render as expected 1`] = `
<div
  style={
    Object {
      "padding": 8,
    }
  }
>
  <ForwardRef(Grid)
    alignContent="center"
    alignItems="center"
    container={true}
    direction="row"
    justifyContent="center"
    spacing={1}
  >
    <ForwardRef(Grid)
      item={true}
      sx={
        Object {
          "display": Object {
            "sm": "none",
            "xs": "block",
          },
        }
      }
      xs={true}
    />
    <ForwardRef(Grid)
      item={true}
      style={
        Object {
          "textAlign": "center",
        }
      }
      xs="auto"
    >
      <ForwardRef(CircularProgress)
        aria-label="Loading"
        color="primary"
        size={18}
        thickness={2}
      />
    </ForwardRef(Grid)>
    <ForwardRef(Grid)
      item={true}
      style={
        Object {
          "textAlign": "center",
        }
      }
      xs="auto"
    >
      <ForwardRef(Typography)
        color="primary"
        component="span"
        style={
          Object {
            "fontSize": "1.33rem",
          }
        }
        variant="h5"
      >
        This is a tst
      </ForwardRef(Typography)>
    </ForwardRef(Grid)>
    <ForwardRef(Grid)
      item={true}
      sx={
        Object {
          "display": Object {
            "sm": "none",
            "xs": "block",
          },
        }
      }
      xs={true}
    />
  </ForwardRef(Grid)>
</div>
`;
